docs/proposals/pod-pid-namespace.md

```diff
@@ -0,0 +1,58 @@
+# Shared PID Namespace for the Docker Runtime
+
+Pods share many namespaces, but the ability to share a PID namespace was not
+supported by Docker until version 1.12. This document proposes how to roll out
+support for sharing the PID namespace in the docker runtime.
+
+## Motivation
+
+Sharing a common PID namespace between containers is desirable for several
+reasons:
+
+  1. It enables signaling between containers, which is useful for side cars
+     (e.g. for signaling a daemon process after rotating logs).
+  2. It eases troubleshooting of pods.
+  3. It allows us to address [Docker's zombie problem][1] by reaping process
+     zombies in the infra container.
+
+## Goals and Non-Goals
+
+Goals include:
+  - Change default behavior in the Kubernetes Docker runtime
+
+Non-goals include:
+  - Creating an init solution that works for all runtimes
+  - Supporting isolated PID namespace indefinitely
+
+## Rollout Plan
+
+Sharing the PID namespace changes an implicit behavior of the Docker runtime
+whereby the command run by the container image is always PID 1. This is a side
+effect of isolated namespaces rather than intentional behavior, but users may
+have built upon this assumption so we should change the default behavior over
+the course of multiple releases.
+
+  1. Release 1.6: Enable the shared PID namespace for pods annotated with
+     `docker.kubernetes.io/shared-pid: true` (i.e. opt-in) when running with
```

Comment:
Interesting, I hadn't seen that proposal. It does make a lot of sense here to use the proposed workflow with the expectation that the field would be removed before coming out of alpha. If it turns out people really want isolated PID namespaces we could instead promote the field to beta, etc.

My hesitation is that, while there's consensus on the motivation, the semantics haven't been determined or documented. Also, since we intend to change the default behavior I'm not sure we want to feature this as a configurable option.

I'd prefer fields over annotations, but I wouldn't choose this as an early adopter of the new process unless you feel strongly.